### DISCUSSION TITLE
Linux | Fedora | Incorrect desktop entry file?

### DISCUSSION THREAD

**Discussion by ddominguez** [ghostty-org/ghostty#3986] [Q&A] [answered]
I can launch ghostty from the terminal, but it doesn’t appear in the application launcher on Fedora. I’m using Fedora Linux 41 (Workstation Edition) and have tried logging out and restarting, but the issue persists.

I installed ghostty using this [build from source method](https://ghostty.org/docs/install/build#installation-directory) `zig build -p $HOME/.local -Doptimize=ReleaseFast`
which installed the following version `info: ghostty version=1.0.1-main+d5703a57`

I was able to locate the desktop entry file for ghostty in `$HOME/.local/share/applications/`
```
[Desktop Entry]
Name=Ghostty
Type=Application
Comment=A terminal emulator
Exec=ghostty
Icon=com.mitchellh.ghostty
Categories=System;TerminalEmulator;
Keywords=terminal;tty;pty;
StartupNotify=true
Terminal=false
Actions=new-window;
X-GNOME-UsesNotifications=true
X-TerminalArgExec=-e
X-TerminalArgTitle=--title=
X-TerminalArgAppId=--class=
X-TerminalArgDir=--working-directory=
X-TerminalArgHold=--wait-after-command

[Desktop Action new-window]
Name=New Window
Exec=ghostty
```

In the same directory I have desktop entry file for kitty and zed
```
[Desktop Entry]
Version=1.0
Type=Application
Name=kitty
GenericName=Terminal emulator
Comment=Fast, feature-rich, GPU based terminal
TryExec=/home/david/.local/kitty.app/bin/kitty
Exec=/home/david/.local/kitty.app/bin/kitty
Icon=/home/david/.local/kitty.app/share/icons/hicolor/256x256/apps/kitty.png
Categories=System;TerminalEmulator;
```
```
[Desktop Entry]
Version=1.0
Type=Application
Name=Zed
GenericName=Text Editor
Comment=A high-performance, multiplayer code editor.
TryExec=/home/david/.local/zed.app/libexec/zed-editor
StartupNotify=true
Exec=/home/david/.local/zed.app/libexec/zed-editor %U
Icon=/home/david/.local/zed.app/share/icons/hicolor/512x512/apps/zed.png
Categories=Utility;TextEditor;Development;IDE;
Keywords=zed;
MimeType=text/plain;application/x-zerosize;x-scheme-handler/zed;
Actions=NewWorkspace;

[Desktop Action NewWorkspace]
Exec=/home/david/.local/zed.app/libexec/zed-editor --new %U
Name=Open a new workspace
```

How does ghostty know which icon to use? I couldn’t locate an icon file named `com.mitchellh.ghostty`.
Both kitty and zed reference an icon within a .app directory. Should ghostty have a .app directory as well? I couldn’t find one.

I was able to locate the icons in `$HOME/.local/share/icons/`
```
/home/david/.local/share/icons/
└── hicolor
    ├── 128x128
    │   └── apps
    │       └── com.mitchellh.ghostty.png
    ├── 128x128@2
    │   └── apps
    │       └── com.mitchellh.ghostty.png
    ├── 16x16
    │   └── apps
    │       └── com.mitchellh.ghostty.png
    ├── 16x16@2
    │   └── apps
    │       └── com.mitchellh.ghostty.png
    ├── 256x256
    │   └── apps
    │       └── com.mitchellh.ghostty.png
    ├── 256x256@2
    │   └── apps
    │       └── com.mitchellh.ghostty.png
    ├── 32x32
    │   └── apps
    │       └── com.mitchellh.ghostty.png
    ├── 32x32@2
    │   └── apps
    │       └── com.mitchellh.ghostty.png
    └── 512x512
        └── apps
            └── com.mitchellh.ghostty.png
```

These are the only icons in this directory.

I also found `share/kio/servicemenus/com.mitchellh.ghostty.desktop` but I don't think this is for [my system.](https://github.com/ghostty-org/ghostty/blob/d5703a57e7d7bcf8dd9bd5711d9f351e5a43446b/build.zig#L663)

**Comment by jcollie** [comment, marked as answer]
Whatever is in the `Exec=` line must be an absolute path, or be findable on the `$PATH`. Gnome won't show the it in the applications menu otherwise. Also sometimes you need to log out and then back in.

**Reply by ddominguez** [reply]
I have already logged out / logged in and restarted.

```
$ command -v ghostty 
/home/david/.local/bin/ghostty
```

I've confirmed that `/home/david/.local/bin/` exists in my $PATH.

**Reply by jcollie** [reply]
I should be more specific, Ghostty needs to be in the _launcher's_ `$PATH`. `~/.local/bin` is usually added to your shell's path by the shell initialization scripts. Try editing the `.desktop` file and add the absolute path to the Ghostty binary.

**Comment by rocketeerbkw** [comment]
I installed the same way and had the same problem on Ubuntu 24.04. The solution is hinted at in the [docs about installation directory](https://ghostty.org/docs/install/build#installation-directory):

> If Ghostty fails to launch using an app icon in your app launcher, ensure that `~/.local/bin` is on your _system_ `PATH`. The desktop environment itself must have that path in the `PATH`. Google for your specific desktop environment and distribution to learn how to do that.

I edited `/etc/environment` and added the full path to my `~/.local/bin`. For example, that would be `/home/david/.local/bin`.
